src/azure-cli/azure/cli/command_modules/acr/custom.py

```diff
@@ -26,14 +26,26 @@
 DEF_DIAG_SETTINGS_NAME_TEMPLATE = '{}-diagnostic-settings'
 SYSTEM_ASSIGNED_IDENTITY_ALIAS = '[system]'
 DENY_ACTION = 'Deny'
-
-
-def acr_check_name(client, registry_name):
-    registry = {
-        'name': registry_name,
-        'type': 'Microsoft.ContainerRegistry/registries'
-    }
-    return client.check_name_availability(registry)
+DOMAIN_NAME_LABEL_SCOPE_UNSECURE = 'Unsecure'
+DOMAIN_NAME_LABEL_SCOPE_RESOURCE_GROUP_REUSE = 'ResourceGroupReuse'
+
+
+def acr_check_name(cmd, client, registry_name, resource_group_name=None, dnl_scope=DOMAIN_NAME_LABEL_SCOPE_UNSECURE):
+    if dnl_scope.lower() == DOMAIN_NAME_LABEL_SCOPE_RESOURCE_GROUP_REUSE.lower() and resource_group_name is None:
+        raise CLIError("Resource group name is required for domain name label scope " +
+                       DOMAIN_NAME_LABEL_SCOPE_RESOURCE_GROUP_REUSE)
```

Comment:
Please use the specific error type instead of `CLIError`
```suggestion
        raise RequiredArgumentMissingError("Resource group name is required for domain name label scope " +
                       DOMAIN_NAME_LABEL_SCOPE_RESOURCE_GROUP_REUSE)
```